### PR TITLE
Define ARM Azure template for the BlobForwarder

### PR DESCRIPTION
**Pull request by tejunior**
This PR creates an ARM (Azure Resource Manager) template to deploy our BlobForwarder function. The customer needs to provide the NR license key for it to operate, a new or existing storage account and finally the data storage container

### CODE REVIEW

File: armTemplates/azuredeploy-blobforwarder.json [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        "Standard_GRS",
+        "Standard_RAGRS",
+        "Standard_ZRS",
+        "Premium_LRS",
+        "Premium_ZRS",
+        "Standard_GZRS",
+        "Standard_RAGZRS"
+      ],
+      "metadata": {
+        "description": "Sku on which to run the Azure Storage account."
```
Comment (noly) [inline armTemplates/azuredeploy-blobforwarder.json:40]: What is Sku? 🤔 it is an acronym? Do we know if Customers are going to understand the reference??


[Here](https://blog.hubspot.com/marketing/inclusive-language) some resources around inclusive language

File: armTemplates/azuredeploy-blobforwarder.json [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      "defaultValue": "StorageV2",
+      "allowedValues": [
+        "Storage",
+        "StorageV2",
+        "BlobStorage",
+        "FileStorage",
+        "BlockBlobStorage"
+      ],
+      "metadata": {
+        "description": "Indicates the type of storage account."
```
Comment (noly) [inline armTemplates/azuredeploy-blobforwarder.json:54]: Could you add more details on why Customers need to indicate this information? Why we need it? Does this setting impact in some how the function trigger?

File: armTemplates/azuredeploy-blobforwarder.json [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      },
+      "minLength": 3,
+      "maxLength": 63
+    }
+  },
+  "variables": {
+    "storageAccountId": "[resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName'))]",
+    "blobForwarderFunctionArtifact": "https://github.com/newrelic/newrelic-azure-functions/releases/latest/download/BlobForwarder.zip",
+    "onePerResourceGroupAndStorageAccountAndContainer": "[uniqueString(resourceGroup().id, parameters('storageAccountName'), parameters('storageAccountContainer'))]",
+    "functionAppName": "[concat('nrlogs-blobforwarder-', variables('onePerResourceGroupAndStorageAccountAndContainer'))]"
```
Comment (noly) [inline armTemplates/azuredeploy-blobforwarder.json:76]: ❔ curiosity, to understand our solution
> onePerResourceGroupAndStorageAccountAndContainer

Let's say one customer want to forward same logs to diff account, unless function is in a diff resource group, is that possible?

File: armTemplates/azuredeploy-blobforwarder.json [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+      "type": "Microsoft.Storage/storageAccounts",
+      "apiVersion": "2021-04-01",
+      "name": "[parameters('storageAccountName')]",
+      "location": "[parameters('storageAccountLocation')]",
+      "sku": {
+        "name": "[parameters('storageAccountSku')]"
+      },
+      "kind": "[parameters('storageAccountKind')]",
+      "properties": {
+        "accessTier": "Hot",
```
Comment (noly) [inline armTemplates/azuredeploy-blobforwarder.json:89]: I'm worried about cost of this storage, maybe we could add some rule to manage it?
https://docs.microsoft.com/en-us/azure/storage/blobs/lifecycle-management-overview

I'm okey also to provide this info in the doc, and let the customer to manage it modifying the template, up to you

**Review by noly — COMMENTED** [review]
Awesome work Teo!!!! 

I left some non-important comments, feel free to ignore them

File: armTemplates/azuredeploy-blobforwarder.json (hunk 1 above)
Comment (tejunior) [reply armTemplates/azuredeploy-blobforwarder.json:40]: Good point, I will use the same parameter azure are using when we create a new service account `Redundancy`

Comment (tejunior) [reply armTemplates/azuredeploy-blobforwarder.json:40]: https://docs.microsoft.com/en-us/azure/templates/microsoft.storage/storageaccounts?tabs=bicep#sku

File: armTemplates/azuredeploy-blobforwarder.json (hunk 3 above)
Comment (tejunior) [reply armTemplates/azuredeploy-blobforwarder.json:76]: No, inside the same resource group it's not possible! Even it was thought as a way to minimize customers mistake and send the same log many times.

File: armTemplates/azuredeploy-blobforwarder.json (hunk 4 above)
Comment (tejunior) [reply armTemplates/azuredeploy-blobforwarder.json:89]: Yes, let's provide docs about it! 👍

Comment (noly) [reply armTemplates/azuredeploy-blobforwarder.json:105]: Agree

**Review by noly — APPROVED** [review]
good-job

**Pull request merged**